### DISCUSSION TITLE
[Vote ended on 2022-08-03] Vote on https://github.com/ansible-community/community-topics/issues/116: normalize booleans in documenation, ansible-lint, ansible-doc, ...?

### DISCUSSION THREAD

**Discussion by felixfontein** [ansible-community/community-topics#120] [Votes done]
Vote on https://github.com/ansible-community/community-topics/issues/116. We aim at normalizing boolean options in documentation, ansible-lint, ansible-doc, AH/GalaxyNG, and other places as follows:

1) `true` and `false`
2) `yes` and `no`
3) `True` and `False`
4) no normalization (basically what we have right now)

Please mention your preferred option as your vote. If you prefer more than one, please give a preference.

Also please note that if we decide on one of options 1) to 3), this does not mean that a) no other possibilities will be mentioned in the documentation anymore, and b) we can force anyone to use these normalized values.

Please review issue #116 before voting so you are informed of the issues with each option.

**Comment by felixfontein** [comment]
My order of preference is 1 > 2 > 4 > 3

**Comment by jillr** [comment]
Order of preference:
1
3
2
4

**Comment by p-rintz** [comment]
My order of preference: 1 > 3 > 2 > 4

**Comment by samccann** [comment]
1 > 2 > 4 > 3

**Comment by briantist** [comment]
1 > 2 > 4 > 3

**Comment by ssbarnea** [comment]
1, 1, 1, 1 -- I like following specs, and we happen to have two that we can follow. Fixing current situation is only about fixing docs, without any code change, as it is not about preventing use of old values, is only about documenting the optimal/standard values. I have no idea if in 2.5 years from now we might remove support for old YAML 1.1 boolean-aliases, but we need to educate people to use safe and future proof values.

**Comment by gotmax23** [comment]
+1 for option 1

**Comment by markuman** [comment]
2 > 4 > 1 > 3

it feels a bit like a similar discussion about linting,  
shall we want to follow a strict yaml linting ruleset in the future?

```yml
extends_documentation_fragment:
    - amazon.aws.aws
    - amazon.aws.ec2
```

vs

```yml
extends_documentation_fragment:
- amazon.aws.aws
- amazon.aws.ec2
```


+1 for option 2

**Comment by russoz** [comment]
Option 1

**Comment by cidrblock** [comment]
Option 1, reasoning in #116

**Comment by KB-perByte** [comment]
+1 for option 1 `true` and `false`

**Comment by sivel** [comment]
I vote for new options called `yes_true` and `no_false`, or maybe `yes_false` and `no_true`.

**Reply by acozine** [reply]
how about working 1/0 into there somehow? I mean `no_0_false` would be stunningly confusing

**Comment by jamescassell** [comment]
2 > 4 > 1 > 3

**Comment by alinabuzachis** [comment]
+1 for option 1. BTW, my order of preference is 1>3>2>4.

**Comment by sean-m-sullivan** [comment]
+1 for option 1. BTW, my order of preference is 1>3>2>4.

module options are lowercase, so are module names, uniformity is best.

**Comment by djdanielsson** [comment]
1>3>2>4 for sure

**Comment by Tompage1994** [comment]
Option 1 has my vote

**Comment by qaxi** [comment]
1 > 2 > 4 > 3

**Comment by amott-rh** [comment]
For me it's 1 then 2. Unnecessary capitalisation is horrible.
While most programmers will understand true/false, the point of ansible is it's easy to read for non-coders and I'd suggest where English is not a first language most people will understand yes or no more readily than true/false. Plus its less typing ;)

**Comment by bcoca** [comment]
Note that  by default ansible-doc output is 'YAML'  (1.1 version as we use pyyaml) but it also outputs JSON (using a switch) and probably other formats in the future, each one has/can have a different standard for 'boolean'. This is irrespective on what values you set on the module documentation as it all gets auto translated by the YAML parser, so if you have `mybool: on` it still appears as `True` or `true` when displayed, so changing the source documentation is mostly futile and irrelevant to the display.   

You also have to consider `ansible-config` as it displays the 'configuration definitions' which can also be considered 'documentation'.

The html pages are controlled by a Jinj2 template which currently displays:
```
{#   Turn boolean values in 'yes' and 'no' values #}
{%   if value['default'] is sameas true %}
{%     set _x = value.update({'default': 'yes'}) %}
{%   elif value['default'] is not none and value['default'] is sameas false %}
{%     set _x = value.update({'default': 'no'}) %}
{%   endif %}
```
This one is in the end the only one we can consistently set to a 'standard' .. but it won't be true for other outputs as that depends on the 'format' being used.

**Comment by mariolenz** [comment]
+1 for option 1

**Comment by branic** [comment]
1 > 3 > 2 > 4

**Comment by NilashishC** [comment]
+1 for option 1, preference: 1>3>2>4.

**Comment by ivarmu** [comment]
+1 for option 1, preference: 1>3>2>4

**Comment by watsonb** [comment]
+1 for option 1, preference: 1>3>2>4

**Comment by Andersson007** [comment]
+1 for option 2
preference: 2>1>3>4

**Comment by apple4ever** [comment]
1 > 3 > 2

(edited because I realized I had it backwards)

**Comment by acozine** [comment]
My order would be 2>1>4>3.

That said, option 2 (normalize to `yes/no`) is how we have been doing it, and folks have complained about the linting errors - that's why we started this discussion. So I am definitely open to 1 (normalize to `true/false`) or 4 (don't normalize at all. Capitalizing just seems silly and distracting, so 3 (normalize to `True/False`) would be my least favorite.

**Comment by pyrodie18** [comment]
Option 1

**Comment by evanstoner** [comment]
2 > 1, definitely not 3 or 4.

**Comment by ganeshrn** [comment]
+1 for option 1

**Comment by priyamsahoo** [comment]
+1 for Option 1!
